### PR TITLE
Use DSN-like strings to define credentials

### PR DESCRIPTION
**Pull request by javiereguiluz**
This is the new best practice for modern Symfony apps. See https://github.com/symfony/recipes/blob/master/symfony/swiftmailer-bundle/2.5/etc/packages/swiftmailer.yaml

### CODE REVIEW

**Comment by javiereguiluz** [comment]
I need someone to explain me why I needed to make this change --> f3d8340 to make this work. Thanks!

**Comment by stof** [comment]
Well, your second commit is because what you defined was not a URL-based DSN for SQLite. It was only a path.
If you want to use it in `url`, use `sqlite://%kernel.project_dir%/...`

**Comment by javiereguiluz** [comment]
@stof that was the problem! I'm an idiot 😊 

However, if I change that, I now see this:

```
Symfony\Component\DependencyInjection\Exception\EnvParameterException:
Incompatible use of dynamic environment variables "DATABASE_URL" found in parameters.

symfony-demo/vendor/symfony/symfony/src/Symfony/Component/DependencyInjection/Dumper/PhpDumper.php:182
symfony-demo/vendor/symfony/symfony/src/Symfony/Component/HttpKernel/Kernel.php:743
symfony-demo/vendor/symfony/symfony/src/Symfony/Component/HttpKernel/Kernel.php:569
symfony-demo/vendor/symfony/symfony/src/Symfony/Component/HttpKernel/Kernel.php:119
symfony-demo/vendor/symfony/symfony/src/Symfony/Bundle/FrameworkBundle/Test/KernelTestCase.php:157
symfony-demo/tests/AppBundle/Command/AddUserCommandTest.php:104
symfony-demo/tests/AppBundle/Command/AddUserCommandTest.php:41
```

**Comment by stof** [comment]
hmm, this message is not clear

**Comment by voronkovich** [comment]
@javiereguiluz, dumper throws an exception because you don't use the `DATABASE_URL` variable in the `test` environment.  See https://github.com/symfony/symfony/blob/v3.3.0/src/Symfony/Component/DependencyInjection/Dumper/PhpDumper.php#L182

**Comment by voronkovich** [comment]
@javiereguiluz, see @dmaicher's comment https://github.com/symfony/symfony-demo/pull/557#issuecomment-300897014

**Comment by javiereguiluz** [comment]
@voronkovich thanks for your comments. That was indeed the error and the proposed fix solved the issue. Thanks!

File: phpunit.xml.dist [hunk 1]
```diff
@@ -10,6 +10,7 @@
     <php>
         <ini name="error_reporting" value="-1" />
         <server name="KERNEL_DIR" value="app/" />
+        <env name="DATABASE_URL" value="sqlite://%kernel.project_dir%/var/data/blog_test.sqlite"/>
```
Comment (stof) [inline phpunit.xml.dist]: I don't think you can use a parameter here, in an actual env var

Comment (dmaicher) [inline phpunit.xml.dist]: Actually you are right but it still works :smile: 

Really weird what happens now when dumping the connection within a test to see the config:

```
  -_params: array:9 [
    "url" => "sqlite://%kernel.project_dir%/var/data/blog_test.sqlite"
    "host" => "%kernel.project_dir%"
    "port" => null
    "user" => "root"
    "password" => null
    "driver" => "pdo_sqlite"
    "driverOptions" => []
    "defaultTableOptions" => []
    "path" => "var/data/blog_test.sqlite"
  ]
```

This seems to be more correct:

```xml
        <env name="DATABASE_URL" value="sqlite:///var/data/blog_test.sqlite" />
```

==>

```
  -_params: array:9 [
    "url" => "sqlite:///var/data/blog_test.sqlite"
    "host" => "localhost"
    "port" => null
    "user" => "root"
    "password" => null
    "driver" => "pdo_sqlite"
    "driverOptions" => []
    "defaultTableOptions" => []
    "path" => "var/data/blog_test.sqlite"
  ]
```

Comment (dmaicher) [inline phpunit.xml.dist]: @javiereguiluz FYI see above

Comment (javiereguiluz) [inline phpunit.xml.dist]: @dmaicher you are right. I fixed that in 464e786 but I think Christophe is right and container params don't work here.

Comment (voronkovich) [inline phpunit.xml.dist]: @javiereguiluz, compiled container doesn't interpolate parameters in environment variables.  See https://github.com/symfony/symfony/blob/v3.3.0/src/Symfony/Component/DependencyInjection/Container.php#L442

**Comment by javiereguiluz** [comment]
Thank you all for your help making this mergeable!

**Pull request closed without merging**